src/storage/database.rs

```diff
@@ -16,18 +16,37 @@ impl<'a> DatabaseBackend<'a> {
         Self { conn }
     }
 
-    pub(super) fn get(&self, path: &str) -> Result<Blob, Error> {
+    pub(super) fn get(&self, path: &str, max_size: usize) -> Result<Blob, Error> {
         use std::convert::TryInto;
 
+        // The maximum size for a BYTEA (the type used for `content`) is 1GB, so this cast is safe:
+        // https://www.postgresql.org/message-id/162867790712200946i7ba8eb92v908ac595c0c35aee%40mail.gmail.com
+        let max_size = max_size.min(std::i32::MAX as usize) as i32;
```

Comment:
Maybe add a debug assertion here to make sure that postgres behaves as expected

Reply:
Hmm, I don't see a way of asserting what the postgres storage limit is. Do you have any idea that doesn't involve writing 1GB of data in the database?

Reply:
I meant asserting that the size we're given matches what we believe will happen, e.g `assert!(max_size <= i32::max_value() as usize);`

Reply:
Well, the line we're commenting on already ensures that... Also, the tests will catch it if for whatever reason it stops working (I needed to add this line *because* tests were failing).